### PR TITLE
0.14

### PR DESCRIPTION
**Pull request by jacobbednarz**
This is the release branch for 0.14.x. 

### Changes:

- [x] All method signatures are now `context` aware and will require an update 
- [x] HTTP verbs have been replaced with `http.Method*` constants
- [x] DNS record attribute `Priority` to *uint16
- [x] DNS record attribute `Proxied` to *bool
- [x] Use Go 1.15 (and test 1.16)

### CODE REVIEW

**Review by patryk — CHANGES_REQUESTED** [review]

File: custom_pages_test.go [hunk 1]
```diff
@@ -264,11 +265,7 @@ func TestUpdateCustomPagesForAccount(t *testing.T) {
 	}
 
 	mux.HandleFunc("/accounts/01a7362d577a6c3019a474fd6f485823/custom_pages/basic_challenge", handler)
-	actual, err := client.UpdateCustomPage(
-		&CustomPageOptions{AccountID: "01a7362d577a6c3019a474fd6f485823"},
-		"basic_challenge",
-		CustomPageParameters{URL: "https://mytestexample.com", State: "customized"},
-	)
+	actual, err := client.UpdateCustomPage(context.TODO(), &CustomPageOptions{AccountID: "01a7362d577a6c3019a474fd6f485823"}, "basic_challenge", CustomPageParameters{URL: "https://mytestexample.com", State: "customized"})
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply custom_pages_test.go:268]: let's cover this in a follow up PR as it's non-breaking and can be completed at any time - raised #597

File: authenticated_origin_pulls.go [hunk 2]
```diff
@@ -53,7 +55,7 @@ func (api *API) SetAuthenticatedOriginPullsStatus(zoneID string, enable bool) (A
 	}{
 		Value: val,
 	}
-	res, err := api.makeRequest("PATCH", uri, params)
+	res, err := api.makeRequestContext(ctx, "PATCH", uri, params)
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply authenticated_origin_pulls.go]: thanks, updated in 6aaf7b8

File: custom_hostname.go [hunk 3]
```diff
@@ -113,12 +115,12 @@ type CustomHostnameFallbackOriginResponse struct {
...
 //
 // API reference: https://api.cloudflare.com/#custom-hostname-for-a-zone-update-custom-hostname-configuration
-func (api *API) UpdateCustomHostnameSSL(zoneID string, customHostnameID string, ssl CustomHostnameSSL) (*CustomHostnameResponse, error) {
+func (api *API) UpdateCustomHostnameSSL(ctx context.Context, zoneID string, customHostnameID string, ssl CustomHostnameSSL) (*CustomHostnameResponse, error) {
 	uri := "/zones/" + zoneID + "/custom_hostnames/" + customHostnameID
 	ch := CustomHostname{
 		SSL: ssl,
 	}
-	res, err := api.makeRequest("PATCH", uri, ch)
+	res, err := api.makeRequestContext(ctx, "PATCH", uri, ch)
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply custom_hostname.go]: done! c684cd6

File: go.mod [hunk 4]
```diff
@@ -3,11 +3,17 @@ module github.com/cloudflare/cloudflare-go
 go 1.14
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply go.mod]: sure thing, 4404182

File: zone_example_test.go [hunk 5]
```diff
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"log"
 
+	"golang.org/x/net/context"
```
Comment (patryk) [inline zone_example_test.go:7]: Different context module?

**Review by patryk — CHANGES_REQUESTED** [review]

File: zone_example_test.go (hunk 5 above)
Comment (jacobbednarz) [reply zone_example_test.go:7]: looks to be similar (if not the same thing).

- https://pkg.go.dev/context
- https://pkg.go.dev/golang.org/x/net/context

different domain perhaps?

i'll remove it either way as stdlib covers this.

Comment (jacobbednarz) [reply zone_example_test.go:7]: 9b62079

Comment (jacobbednarz) [reply zone_example_test.go]: ah, the second is deprecated since 1.7 as it's now in stdlib. i wonder why this got pulled in 🤔

**Review by patryk — APPROVED** [review]

**Pull request merged**